etl/steps/data/garden/wb/2024-01-17/world_bank_pip.py

```diff
@@ -0,0 +1,1047 @@
+
+    # Add variables below first poverty line to the stacked variables
+    col_stacked_n.append(f"headcount_{povlines[0]}")
+    col_stacked_pct.append(f"headcount_ratio_{povlines[0]}")
+
+    # Calculate stacked variables which "jump" the original order
+
+    tb[f"headcount_between_{povlines[1]}_{povlines[4]}"] = (
+        tb[f"headcount_{povlines[4]}"] - tb[f"headcount_{povlines[1]}"]
+    )
+    tb[f"headcount_between_{povlines[4]}_{povlines[6]}"] = (
+        tb[f"headcount_{povlines[6]}"] - tb[f"headcount_{povlines[4]}"]
+    )
+
+    tb[f"headcount_ratio_between_{povlines[1]}_{povlines[4]}"] = (
+        tb[f"headcount_ratio_{povlines[4]}"] - tb[f"headcount_ratio_{povlines[1]}"]
+    )
+    tb[f"headcount_ratio_between_{povlines[4]}_{povlines[6]}"] = (
+        tb[f"headcount_ratio_{povlines[6]}"] - tb[f"headcount_ratio_{povlines[4]}"]
+    )
+
+    return tb, col_stacked_n, col_stacked_pct
+
+
+def calculate_inequality(tb: Table) -> Table:
+    """
+    Calculate inequality measures: decile averages and ratios
+    """
+
+    col_decile_share = []
+    col_decile_avg = []
+    col_decile_thr = []
+
+    for i in range(1, 11):
+        if i != 10:
+            varname_thr = f"decile{i}_thr"
+            col_decile_thr.append(varname_thr)
+
+        varname_share = f"decile{i}_share"
+        varname_avg = f"decile{i}_avg"
+        tb[varname_avg] = tb[varname_share] * tb["mean"] / 0.1
+
+        col_decile_share.append(varname_share)
+        col_decile_avg.append(varname_avg)
+
+    # Multiplies decile columns by 100
+    tb.loc[:, col_decile_share] = tb[col_decile_share] * 100
+
+    # Create bottom 50 and middle 40% shares
+    tb["bottom50_share"] = (
+        tb["decile1_share"] + tb["decile2_share"] + tb["decile3_share"] + tb["decile4_share"] + tb["decile5_share"]
+    )
+    tb["middle40_share"] = tb["decile6_share"] + tb["decile7_share"] + tb["decile8_share"] + tb["decile9_share"]
+
+    # Palma ratio and other average/share ratios
+    tb["palma_ratio"] = tb["decile10_share"] / (
+        tb["decile1_share"] + tb["decile2_share"] + tb["decile3_share"] + tb["decile4_share"]
+    )
+    tb["s80_s20_ratio"] = (tb["decile9_share"] + tb["decile10_share"]) / (tb["decile1_share"] + tb["decile2_share"])
+    tb["p90_p10_ratio"] = tb["decile9_thr"] / tb["decile1_thr"]
+    tb["p90_p50_ratio"] = tb["decile9_thr"] / tb["decile5_thr"]
+    tb["p50_p10_ratio"] = tb["decile5_thr"] / tb["decile1_thr"]
+
+    # Replace infinite values with nulls
+    tb = tb.replace([np.inf, -np.inf], np.nan)
+    return tb
+
+
+def identify_rural_urban(tb: Table) -> Table:
+    """
+    Amend the entity to reflect if data refers to urban or rural only
+    """
+
+    # Make country and reporting_level columns into strings
+    tb["country"] = tb["country"].astype(str)
+    tb["reporting_level"] = tb["reporting_level"].astype(str)
+
+    tb.loc[(tb["reporting_level"].isin(["urban", "rural"])), "country"] = (
```

Comment:
Nit - when I use an index multiple times in an expression, I tend to DRY it with
```
ix = tb["reporting_level"].isin(["urban", "rural"])
tb.loc[ix, 'country'] = tb.loc[ix, 'country'] + " (" + tb.loc[ix, 'reporting_level'] + ")"
```
It's a bit more readable